### PR TITLE
Backports/2.1.4

### PR DESCRIPTION
**Pull request by 1Dragoon**
Hopefully I'm doing this correctly:

* Fix `impl SqlOrd` postgres > postgres_backend feature flag.
* Allow `Queryable` to be used with multiple table names
* Fix chrono deprecation warnings
* Adjusted some unit tests to reflect chrono changes

Note: One unit test is failing, but I'm uncertain whether it's caused by my local build environment. It was already failing without any of these changes.

```
failures:

---- pg::expression::extensions::interval_dsl::tests::intervals_match_pg_values_f64 stdout ----
thread 'pg::expression::extensions::interval_dsl::tests::intervals_match_pg_values_f64' panicked at /home/1Dragoon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/quickcheck-1.0.3/src/tester.rs:165:28:
[quickcheck] TEST FAILED. Arguments: (-2578.740551172832)
```

### CODE REVIEW

File: diesel/src/pg/types/date_and_time/chrono.rs [hunk 1]
```diff
@@ -281,7 +281,7 @@ mod tests {
         assert!(query.get_result::<bool>(connection).unwrap());
 
         let max_date = NaiveDate::MAX;
-        let query = select(sql::<Date>("'262143-12-31'::date").eq(max_date));
+        let query = select(sql::<Date>("'262142-12-31'::date").eq(max_date));
```
Comment (1Dragoon) [inline diesel/src/pg/types/date_and_time/chrono.rs:284]: According to the Chrono documentation, NaiveDate::MAX is 262142. I'm guessing this was part of a recent change?

File: Cargo.toml [hunk 2]
```diff
@@ -26,6 +26,7 @@ members = [
     "examples/sqlite/getting_started_step_2",
     "examples/sqlite/getting_started_step_3",
 ]
+resolver = "2"
```
Comment (1Dragoon) [inline Cargo.toml:29]: Clippy was complaining that resolver was version 1, which apparently doesn't work with 2021 edition, so it was warning to switch to 2.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,13 +3,20 @@
 All user visible changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/), as described
 for Rust libraries in [RFC #1105](https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md)
-For any named minimal supported Rust version we guarantee that it is possible to build Diesel with the 
-default features enabled using some set of dependencies. Those set of dependencies is not necessarily 
-an up to date version of the specific dependency. We check this by using the unstable `-Z minimal-version` cargo flag. 
+For any named minimal supported Rust version we guarantee that it is possible to build Diesel with the
```
Comment (1Dragoon) [inline CHANGELOG.md:6]: The vscode markdown extension suggested some changes, so I let it autofix. Let me know these are not desired.

**Comment by 1Dragoon** [comment]
So this is odd, clippy is tossing lints about `chrono` deprecated methods, yet `chrono` 0.4.20 doesn't have the replacement methods. Bumping `chrono` to anything higher (even from 0.4.20 to 0.4.21) results in a `libc` conflict in the compile tests, which can't be resolved because `r2d2` has a `parking_lot` dependency that requires a lower version than what `chrono` and `time` require.

**Comment by weiznich** [comment]
Thanks for opening this PR. 
For the chrono issues I would prefer to pull in the relevant changes from the master branch:

https://github.com/diesel-rs/diesel/pull/3953
and
https://github.com/diesel-rs/diesel/commit/4c7b6ea61705ef18c3465354d4d1e9d925661b0c

That should hopefully resolve those issues without changing the minimal supported chrono version and prevent that breaking change.

**Comment by 1Dragoon** [comment]
Sure, merged those in. The CI is failing here still:

image

I've seen this before. In my local setup I had to fix this by modifying ./bin/test to specify the db type as a feature directly, maybe the CI needs to be adjusted somehow?

File: Cargo.toml (hunk 2 above)
Comment (weiznich) [reply Cargo.toml:29]: That's what is causing that last CI issues. Seems like we should rather pull in https://github.com/diesel-rs/diesel/commit/b816eda5325daf27162ad1614ddbfc2e609cdfce to fix that.

**Comment by 1Dragoon** [comment]
No matter what I do I can't seem to get the MSRV to pass

**Comment by weiznich** [comment]
I pushed another commit that hopefully fixes the MSRV build. Will try to do a final review and a release tomorrow.

**Review by weiznich — APPROVED** [review]

**Comment by weiznich** [comment]
Published as https://github.com/diesel-rs/diesel/releases/tag/v2.1.5

Thanks again for your work on the backports.

**Pull request merged**